### ISSUE TITLE
Can't make predictions with served TensorFlow (Keras) model.

### ISSUE THREAD

**Issue by SansDK** [tensorflow/serving#1296]
### System information
I'm running the latest TensorFlow serving docker image.

### Describe the problem
The problem that I have occurs when I try to make a HTTP POST request to the TF Serving container. The error I get is:
`"error": "Failed to process element: 0 key: key_name of \'instances\' list. Error: Invalid argument: JSON object: does not have named input: key_name"`

I exported my Keras model using the following code:
```
tf.keras.backend.set_learning_phase(0)
with tf.keras.backend.get_session() as sess:
tf.saved_model.simple_save(
        sess,
        export_path,
        inputs={'input_request': model.input},
        outputs={t.name: t for t in model.outputs})
```

I use the Python 'requests' library to make a POST request to the model. The format of my request is:
`{"instances": [{"example1":1, "example2": 2}]}`

Why do I get this error and how can I fix it?

**Comment by hgadig** [comment]
The following StackOverflow answer addressed a similar issue. Please take a look at this.

https://stackoverflow.com/questions/52208668/tensorflow-serving-error-invalid-argument-json-object-does-not-have-named-inp

**Comment by SansDK** [comment]
I tried a few suggestions from the StackOverflow answers.
I tried inputs with the following formats:

- {"instances": {["col1": 0.0, "col2":1.1]}} --> error mentioned in question
- {"instances": {"col1": 0.0, "col2":1.1}}  --> error: Excepting \'instances\' to be an list/array"
- {"col1": 0.0, "col2":1.1} --> error: Missing key: instances or inputs

Also, I tried making a 'regress' call instead of a predict call, but this gave me the error: "Expected regression signature method_name to be tensorflow/serving/regress. Was: tensorflow/serving/predict"

I'm starting to suspect that the problem lies in the exporting part of the Keras model, but I'm not sure.. Any help?

**Comment by SansDK** [comment]
Okay, I fixed it.
There were a few problems with my usage.
First of all, in my export code I define:

`inputs={'input_request': model.input}`

the output of model.input is:
`<tf.Tensor 'dense_1_input:0' shape=(?, 282) dtype=float32>`

Which got me thinking that I should pass an array with shape (1, 282) (single prediction in this case) to the TF serving http api. This array input is named 'input_request', because I defined it like that in the export code.

My final (working) json request looks something like this:
`{"instances":[{"input_request":[0.0,0.1,0.2,0.3]}]}`

0.0 , 0.1 , etc. being the values for every single feature.

Thanks.
